lib/Backend/Lower.cpp

```diff
@@ -9519,12 +9519,17 @@ Lowerer::GenerateFastBrBReturn(IR::Instr * instr)
         Js::OpCode::BrNeq_A, labelHelper, instr);
 
     // MOV cachedDataOpnd, forInEnumeratorOpnd->enumerator.cachedData
+    // CMP cachedDataOpnd, nullptr
+    // JEQ $helper
     // MOV enumeratedCountOpnd, forInEnumeratorOpnd->enumerator.enumeratedCount
     // CMP enumeratedCountOpnd, cachedDataOpnd->cachedCount
     // JLT $loopBody
     IR::RegOpnd * cachedDataOpnd = IR::RegOpnd::New(TyMachPtr, m_func);
     InsertMove(cachedDataOpnd,
         GetForInEnumeratorFieldOpnd(forInEnumeratorOpnd, Js::ForInObjectEnumerator::GetOffsetOfEnumeratorCachedData(), TyMachPtr), instr);
+
+    InsertCompareBranch(cachedDataOpnd, IR::AddrOpnd::NewNull(m_func), Js::OpCode::BrEq_A, labelHelper, instr);
```

Comment:
Can you just set "canUseJitFastPath" to false do you don't need to have addition check here?
There is also the cachedDataType you can use to invalidate the cache maybe?